index.html

```diff
@@ -232,39 +235,393 @@ <h3>Create</h3>
+signatures. An <em>identifier</em>, <em>multibase value</em>, and
+<em>options</em> are the required inputs. A <em>verification method</em> is
+the resulting output.
+          </p>
+
+          <ol class="algorithm">
+            <li>
+Initialize <em>verification method</em> to an empty object.
+            </li>
+            <li>
+Set <em>multicodec value</em> and <em>raw public key bytes</em> to the result
+of passing <em>multibase value</em> and <em>options</em> to
+<a href="#decode-public-key-algorithm"></a>.
+            </li>
+            <li>
+Ensure the proper key length of <em>raw public key bytes</em> based on the
+<em>multicodec value</em> table provided below:
+              <table class="simple">
+                <thead>
+                  <th>Multicodec hexadecimal value</th>
+                  <th>public key byte length</th>
+                  <th>Description</th>
+                </thead>
+
+                <tbody>
+                  <tr>
+                    <td>`0xe7`</td>
+                    <td>33 bytes</td>
+                    <td>secp256k1-pub - Secp256k1 public key (compressed)</td>
+                  </tr>
+                  <tr>
+                    <td>`0xec`</td>
+                    <td>32 bytes</td>
+                    <td>x25519-pub - Curve25519 public key</td>
+                  </tr>
+                  <tr>
+                    <td>`0xed`</td>
+                    <td>32 bytes</td>
+                    <td>ed25519-pub - Ed25519 public key</td>
+                  </tr>
+                  <tr>
+                    <td>`0x1200`</td>
+                    <td>33 bytes</td>
+                    <td>p256-pub - P-256 public key (compressed)</td>
+                  </tr>
+                  <tr>
+                    <td>`0x1201`</td>
+                    <td>?? bytes</td>
+                    <td>p384-pub - P-384 public key (compressed)</td>
+                  </tr>
+                  <tr>
+                    <td>`0x1202`</td>
+                    <td>?? bytes</td>
+                    <td>p521-pub - P-521 public key (compressed)</td>
+                  </tr>
+                  <tr>
+                    <td>`0x1205`</td>
+                    <td>?? bytes</td>
+                    <td>
+rsa-pub - RSA public key. DER-encoded ASN.1 type RSAPublicKey according to
+IETF RFC 8017 (PKCS #1)
+                    </td>
+                  </tr>
+                </tbody>
+              </table>
+If the byte length of <em>raw public key bytes</em> does not match the expected
+public key length for the associated <em>multicodec value</em>, an
+INVALID_PUBLIC_KEY_LENGTH error MUST be raised.
+            </li>
+            <li>
+Set the <em>verification method</em> `id` value by concatenating
+<em>identifier</em>, a hash character (`#`), and the <em>multicodec value</em>.
+            </li>
+            <li>
+Set the <em>key encoding type</em> value to the <em>options</em> `type` value
```

Comment:
Am i reading this that `MultiKey` is the default type? It seems like did key would have highest degree of interop with off the shelf tooling if `JsonWebKey` were the default key type... since there are far more libraries that support JWK than that support non standard key types.

Reply:
The implementing software library has the option of picking what the default type is. This line is there to ensure that if the software library doesn't set a default type, that one is asserted. I was trying to avoid the exact issue that you're driving at -- a fight over what the defaults should be. Rather than have that fight, it's up to the software implementer to pick reasonable defaults for their library by defaulting `key encoding type` to whatever they think is best. Transmute will pick JsonWebKey, DB will pick Multikey.

Perhaps I should add some language describing this, or change the algorithm to say that `key encoding type` MUST be set to a default value by the implementation, or an error must be thrown?

Reply:
I think this is a fight that needs to be resolved to protect the method. 
I feel strongly that a default that is not well supported will harm the method.
I like the language asserting that a default must exist, and that it should be `MultiKey`... except that it says it should be `MultiKey`.

Reply:
If you like, we can move it to an issue, like you did for JWK production rules.

Reply:
> I think this is a fight that needs to be resolved to protect the method.

Ugh, no thanks. :)

I've removed the `Multikey` default and suggested that implementations should pick a reasonable default. If they don't pick a reasonable default, the algorithm throws an error now:

Addressed in 1250253de7506288bc645ff8c2a8a693461369e0

Reply:
> The implementing software library has the option of picking what the default type is. This line is there to ensure that if the software library doesn't set a default type, that one is asserted. I was trying to avoid the exact issue that you're driving at -- a fight over what the defaults should be. Rather than have that fight, it's up to the software implementer to pick reasonable defaults for their library by defaulting `key encoding type` to whatever they think is best. Transmute will pick JsonWebKey, DB will pick Multikey.
> 
https://www.npmjs.com/package/key-did-resolver uses JsonWebKey2020
e.g. https://github.com/ceramicnetwork/js-ceramic/blob/develop/packages/key-did-resolver/src/secp256r1.ts#L27-L52